core/src/main/java/software/amazon/awssdk/core/retry/conditions/OrThrottlingCondition.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.core.retry.conditions;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import software.amazon.awssdk.annotations.SdkPublicApi;
+
+/**
+ * Composite retry condition that evaluates to true if any containing condition evaluates to true.
+ */
+@SdkPublicApi
+public class OrThrottlingCondition implements ThrottlingCondition {
+
+    private List<ThrottlingCondition> conditions = new ArrayList<>();
+
+    public OrThrottlingCondition(ThrottlingCondition... conditions) {
+        Collections.addAll(this.conditions, conditions);
+    }
+
+    /**
+     * @return True if any condition returns true. False otherwise.
+     */
+    @Override
+    public boolean isThrottlingException(Exception context) {
```

Comment:
Same as prior comment

Reply:
do you mean create something like `RetryOnThrottlingCondition` to implement `RetryConditon`? there are places where we actually check if it's a throttling exception  and not if it should retry though.

https://github.com/aws/aws-sdk-java-v2/blob/AwsSpecificRetryCondition/core/src/main/java/software/amazon/awssdk/core/retry/RetryHandler.java#L56